### DISCUSSION TITLE
Question about H2 board versions and component compatibility

### DISCUSSION THREAD

**Discussion by Xanuman** [kachurovskiy/nanoels#214] [Q&A] [answered]
Hello, Maxim!
Thank you for this wonderful open source project!

There are two versions of the H2 guitar board:
1) Old, simplified from 2022.08.15
2) Newer from 2023.04.04

Questions:
- How important is it to use the new version? Are there any critical bug fixes?
- Have people already made the new version in hardware? Is it working?

I plan to install your H2 system on the CJ0618 machine
Already ordered:
- Stepper motor NEMA23 57hs112
- Driver DM556
- Optical encoder E6B2CWZ6C (1800 PR)

I have not made any mistakes, will these components work correctly with the H2 guitar?

**Comment by kachurovskiy** [comment, marked as answer]
Hi @Xanuman, thank you. Both versions of the board work well. I suggest to use the simpler, older version. Newer version of the board was created for CE/FCC compliance reasons.

⚠️ Due to Arduino relatively low frequency and realtime requirements (can't plan the motion ahead) H2 can only issue 2800 step impulses per second. This means that the motor can reach ~800 RPM at 200 steps per revolution or ~200 RPM at 800 steps per revolution. Make sure to use a motor with a driver that can do full steps or use the [H4 controller](https://github.com/kachurovskiy/nanoels/tree/main/h4) instead.

So if you're going to proceed with the H2 build, make sure that your driver has 200 step mode. Some only support 800 steps and above.
